practical_09/taxi_simulator.py

```diff
@@ -0,0 +1,96 @@
+"""
+Taxi simulator to allow customer to drive a taxi of their choosing
+"""
+
+from practical_09.silver_service_taxi import SilverServiceTaxi
+from practical_09.taxi import Taxi
+
+MENU = """
+q)uit, c)hoose, d)rive
+>>> """
+
+taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+
+
+def main():
+    """Present user with a list of available taxis.
+    Allow them to drive that taxi a desired distance.
+    Display cumulative bill. Repeat until user quits."""
+    total_bill = 0
+    user_choice = input(MENU)
+    while user_choice.lower() != "q":
+        if user_choice.lower() == "c":
+            taxi_choice = choose_taxi()
+        elif user_choice.lower() == "d":
+            try:
+                total_bill = drive_taxi(taxi_choice, total_bill)
+            except UnboundLocalError:
+                print("Choose a taxi before driving")
+        else:
+            print("Invalid choice")
+        print(f"Bill to date: ${total_bill:0.2f}")
+        user_choice = input(MENU)
+    print(f"Total trip cost: ${total_bill:0.2f}")
+    print("Taxis are now:")
+    for taxi in taxis:
+        print(taxi)
```

Comment:
Do you think this code looks like display_taxis()? It's almost the same. 
What you have now is not a problem though, just food for thought.